src/Facade/Common/AsyncFacadeService.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<IResult<TResource>> Add(
+            TResource resource, 
+            IEnumerable<string> privileges = null, 
+            int? userNumber = null)
+        {
+            T entity;
+
+            var privilegesList = privileges?.ToList();
+
+            try
+            {
+                entity = await this.CreateFromResourceAsync(resource, privilegesList);
+            }
+            catch (DomainException exception)
+            {
+                return new BadRequestResult<TResource>(exception.Message);
+            }
+
+            await this.repository.AddAsync(entity);
+
+            if (userNumber.HasValue)
+            {
+                await this.MaybeSaveLog("Create", userNumber, entity, resource, default);
+            }
+
+            await this.transactionManager.CommitAsync();
+
+            return new CreatedResult<TResource>(this.BuildResource(entity, privilegesList));
+        }
+
+        public async Task<IResult<TResource>> Update(
+            TKey id, 
+            TUpdateResource updateResource, 
+            IEnumerable<string> privileges = null,
+            int? userNumber = null)
+        {
+            var entity = await this.FindById(id);
+            if (entity == null)
+            {
+                return new NotFoundResult<TResource>();
+            }
+
+            var privilegesList = privileges?.ToList();
+
+            try
+            {
+                await this.UpdateFromResourceAsync(entity, updateResource, privilegesList);
+            }
+            catch (DomainException exception)
+            {
+                return new BadRequestResult<TResource>($"Error updating {id} - {exception.Message}");
+            }
+
+            if (userNumber.HasValue)
+            {
+                await this.MaybeSaveLog("Update", userNumber, entity, default, updateResource);
+            }
+
+            await this.transactionManager.CommitAsync();
+
+            return new SuccessResult<TResource>(this.BuildResource(entity, privilegesList));
+        }
+
+        public Task<IResult<TResource>> GetApplicationState(
+            IEnumerable<string> privileges = null)
+        {
+            // todo
+            throw new NotImplementedException();
+        }
+
+        public Task<IResult<TResource>> DeleteOrObsolete(
+            TKey id, 
+            IEnumerable<string> privileges = null, 
+            int? userNumber = null)
+        {
+            // todo
+            throw new NotImplementedException();
+        }
+
+        public Task<IResult<IEnumerable<TResource>>> UpdateCollection(
+            IEnumerable<UpdateCollectionResource<TResource, TUpdateResource, TKey>> updates, 
+            IEnumerable<string> privileges = null, 
+            int? userNumber = null)
+        {
+            // todo
+            throw new NotImplementedException();
+        }
```

Comment:
don't need any of these yet so didn't flesh out async implementations  - would be a good excercise for someone else to try too